src/PoolManager.sol

```diff
@@ -275,11 +282,11 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         if (currency.isNative()) {
             paid = msg.value;
         } else {
-            uint256 reservesBefore = reservesOf[currency];
-            reservesOf[currency] = currency.balanceOfSelf();
-            paid = reservesOf[currency] - reservesBefore;
+            uint256 reservesBefore = Reserves.get(currency);
+            if (reservesBefore == 0) revert ReservesMustBeSynced();
```

Comment:
would be great to see a test for calling settle without calling sync first

Reply:
I added these tests. I think there are likely more invariant/integ tests but either can followup after this (add to testing doc) or include here. up to you. would love another look though!